File: BE/src/main/java/com/siupo/restaurant/service/address/AddressService.java
package com.siupo.restaurant.service.address;

import com.siupo.restaurant.dto.request.AddressRequest;
import com.siupo.restaurant.dto.response.AddressResponse;
import com.siupo.restaurant.model.User;

import java.util.List;

public interface AddressService {
    List<AddressResponse> getAddresses(User user);
    AddressResponse addAddress(User user, AddressRequest address);
    AddressResponse updateAddress(User user, Long addressId, AddressRequest address);
    void deleteAddress(User user, Long addressId);
    AddressResponse getAddressDefault(User user);
    AddressResponse setAddressDefault(User user, Long addressId);
}

File: FE-customer/src/services/addressService.ts
import addressApi from "../api/addressApi";
import type { Address } from "../types/models/address";
import type { AddressUpdateRequest } from "../types/requests/address-update.request";

const addressService = {
  getDefaultAddress: async () => {
    const res = await addressApi.getDefaultAddress();
    return res;
  },

  setDefaultAddress: async (address: Address) => {
    const res = await addressApi.setDefaultAddress(address.id!);
    return res;
  },

  getAddresses: async () => {
    const res = await addressApi.getAddresses();
    return res;
  },

  addAddress: async (data: Address) => {
    const res = await addressApi.addAddress(data);
    return res;
  },

  updateAddress: async (data: AddressUpdateRequest) => {
    const res = await addressApi.updateAddress(data);
    return res;
  },

  deleteAddress: async (addressId: number) => {
    const res = await addressApi.deleteAddress(addressId);
    return res;
  },
};

export default addressService;
